src/components/molecules/NavBar/components/NavBarLogin/NavBarLogin.tsx

```diff
@@ -1,28 +1,38 @@
-import React, { FC, useCallback } from "react";
-import { useHistory } from "react-router-dom";
+import { FC, useCallback } from "react";
 
-import { DEFAULT_SPACE_SLUG, DEFAULT_WORLD_SLUG } from "settings";
+import {
+  ATTENDEE_SPACE_ENTRANCE_URL,
+  ATTENDEE_SPACE_INSIDE_URL,
+} from "settings";
 
-import { venueEntranceUrl } from "utils/url";
+import { generateUrl } from "utils/url";
 
 import { useSpaceParams } from "hooks/spaces/useSpaceParams";
 
 import "./NavBarLogin.scss";
 
-export const NavBarLogin: FC = () => {
-  const history = useHistory();
+export interface NavBarLoginProps {
+  hasEntrance: boolean;
+}
+
+export const NavBarLogin: FC<NavBarLoginProps> = ({ hasEntrance }) => {
   const { worldSlug, spaceSlug } = useSpaceParams();
 
-  const navigateToDefault = useCallback(
-    () =>
-      history.push(
-        venueEntranceUrl(
-          worldSlug ?? DEFAULT_WORLD_SLUG,
-          spaceSlug ?? DEFAULT_SPACE_SLUG
-        )
-      ),
-    [history, worldSlug, spaceSlug]
-  );
+  const navigateToDefault = useCallback(() => {
+    const redirectUrl = generateUrl({
+      route: !hasEntrance
+        ? ATTENDEE_SPACE_INSIDE_URL
+        : ATTENDEE_SPACE_ENTRANCE_URL,
+      required: ["worldSlug", "spaceSlug"],
+      params: { worldSlug, spaceSlug, step: "1" },
```

Comment:
Goran added a const `export const DEFAULT_ENTER_STEP = "1";` in `src/settings/urlSettings.ts`
https://github.com/sparkletown/sparkle/pull/2718/files#diff-4412f8f75290a96968c8717d422435457d190e4f557c135122f1ce728de4a430
we might as well do the same here and resolve conflicts later

Reply:
Used in both places.